habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/migration/CustomPoetrycoreVersionMigration.java

```diff
@@ -159,4 +176,68 @@ private boolean isPoetrycoreUpgradeRequired(Map.Entry<String, Object> dependency
         }
         return false;
     }
+
```

Comment:
S: In other places, we use semver4j and it handles a lot of this complexity and probably will cover a lot of edges too: https://github.com/vdurmont/semver4j

Reply:
Used semver4j and cleaned up a lot of the code.